### DISCUSSION TITLE
Integration across genotype and batch

### DISCUSSION THREAD

**Discussion by ajebrz** [satijalab/seurat#8674] [Q&A]
Hi!
Thanks a lot to the Satijalab team for making such a great and intuitive tool for single cell analysis. I'm new to scRNAseq so would really appreciate some help:)
I have more of a conceptual question regarding integration of multiple "factors". 
I'm analysing a dataset of immune cells, which have been isolated from murine lymph nodes. I have two genotypes: disease and WT, each genotype has 4 mice (2 male and 2 female). I've integrated the data across genotype using Seurat's integration tool (CCA, Seurat version 4.4.0).
Recently we have noticed a relatively strong batch effect in the integrated data, where the 4 mice that were handled before noon (1 WT + 3 disease; 2 males + 2 females) seem to have fewer cells in some clusters and more cells in others, as compared to mice handled in the afternoon (3 WT + 1 disease, 2 males + 2 females). This is already very noticable for one of the big clusters (cluster 2, with 16.696 cells, where 90% of cells stem from mice in one of the batches).
When comparing factors: genotype and batch using PCA plots, cells separate first by batch, then genotype. I'm not really sure how to proceed from here. Are there any integration tools that would allow me to introduce batch as a "covariate", so that both genotype and batch can be taken into concideration by the integration algorithm?
Here I'm thinking of something in the vain of covariates that you can provide to regression models for DE analysis (DESeq2). Any help much appreciated!!

Best,
Adrianna

**Comment by caitlinaamodt** [comment]
The recommendation is to split the layers by all of the covariates in this response. https://github.com/satijalab/seurat/issues/8136
